hdk-rust/src/macros.rs

```diff
@@ -11,7 +12,87 @@ macro_rules! load_json {
     }};
 }
 
-/// A macro for describing zomes
+/// Every Zome must utilize the `define_zome`
+/// macro in the main library file in their Zome.
+/// The `define_zome` macro has 3 component parts:
+/// 1. entries: an array of [ValidatingEntryType](entry_definition/struct.ValidatingEntryType.html) as returned by using the [entry](macro.entry.html) macro
+/// 2. genesis: `genesis` is a callback called by Holochain to every Zome implemented within a DNA.
+///     It gets called when a new agent is initializing an instance of the DNA for the first time, and
+///     should return `Ok` or an `Err`, depending on whether the agent can join the network or not.
+/// 3. functions: `functions` is divided up into `capabilities`, which specify who can access those functions.
+///     `functions` must be a tree structure where the first children are `capabilities`
+///     and the children of those `capabilities` are actual function definitions.
+/// # Examples
+///
+/// ```rust
+/// #[macro_use]
+/// extern crate hdk;
+/// extern crate serde;
+/// #[macro_use]
+/// extern crate serde_derive;
+/// #[macro_use]
+/// extern crate serde_json;
+/// extern crate boolinator;
+///
+/// #[derive(Serialize, Deserialize)]
+/// pub struct Post {
+///     content: String,
+///     date_created: String,
+/// }
+///
+/// fn handle_hash_post(content: String) -> serde_json::Value {
+///     let maybe_address = hdk::hash_entry("post", json!({
+///         "content": content,
+///         "date_created": "now"
+///     }));
+///     match maybe_address {
+///         Ok(address) => {
+///             json!({"address": address})
+///         }
+///         Err(hdk_error) => hdk_error.to_json(),
+///     }
+/// }
+///
+/// define_zome! {
+///     entries: [
+///         entry!(
```

Comment:
It's a little confusing that the `entry!` example is duplicated here, and above is inside the `definitions()` function.  I think that may create confusion without a tiny bit of text some place, but no big deal.